core/src/proxy/block_engine_stage.rs

```diff
@@ -101,32 +105,59 @@ impl BlockEngineStage {
         exit: Arc<AtomicBool>,
         block_builder_fee_info: &Arc<Mutex<BlockBuilderFeeInfo>>,
     ) -> Self {
-        let block_builder_fee_info = block_builder_fee_info.clone();
-
-        let thread = Builder::new()
-            .name("block-engine-stage".to_string())
-            .spawn(move || {
-                let rt = tokio::runtime::Builder::new_multi_thread()
-                    .enable_all()
-                    .build()
-                    .unwrap();
-                rt.block_on(Self::start(
-                    block_engine_config,
-                    cluster_info,
-                    bundle_tx,
-                    packet_tx,
-                    banking_packet_sender,
-                    exit,
-                    block_builder_fee_info,
-                ));
-            })
-            .unwrap();
+        let handles = std::iter::once(Either::Left(block_engine_config))
+            .chain(secondary_urls.into_iter().map(Either::Right))
+            .enumerate()
+            .map(|(index, config)| {
+                let thread_name = match &config {
+                    Either::Left(_) => "block-engine-primary".to_string(),
+                    Either::Right(_) => format!("block-engine-secondary-{}", index),
+                };
+
+                Self::spawn_block_engine_thread(
+                    thread_name,
+                    config,
+                    cluster_info.clone(),
+                    bundle_tx.clone(),
+                    packet_tx.clone(),
+                    banking_packet_sender.clone(),
+                    exit.clone(),
+                    block_builder_fee_info.clone()
+                )
+            }).collect::<Vec<_>>();
 
         Self {
-            t_hdls: vec![thread],
+            t_hdls: handles,
         }
     }
 
+    fn spawn_block_engine_thread(
+        thread_name: String,
+        config: Either<Arc<Mutex<BlockEngineConfig>>, String>,
+        cluster_info: Arc<ClusterInfo>,
+        bundle_tx: Sender<Vec<PacketBundle>>,
+        packet_tx: Sender<PacketBatch>,
+        banking_packet_sender: BankingPacketSender,
+        exit: Arc<AtomicBool>,
+        block_builder_fee_info: Arc<Mutex<BlockBuilderFeeInfo>>,
+    ) -> JoinHandle<()>{
+        Builder::new().name(thread_name).spawn(move ||{
+            let rt = tokio::runtime::Builder::new_multi_thread()
+                .enable_all()
+                .build()
+                .unwrap();
+            rt.block_on(Self::start(
+                config,
+                cluster_info,
+                bundle_tx,
+                packet_tx,
+                banking_packet_sender,
+                exit,
+                block_builder_fee_info,
+            ));
+        }).unwrap()
```

Comment:
Now that we have a single callsite lets just inline this to reduce line count & need to redeclare all variables which is painful and error prone